lib/translator/unmap.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "unmap.h"
+
+namespace translator {
+
+// Section 5.6
+// https://www.nvmexpress.org/wp-content/uploads/NVM-Express-SCSI-Translation-Reference-1_1-Gold.pdf
+StatusCode UnmapToNvme(absl::Span<const uint8_t> scsi_cmd,
+                       absl::Span<const uint8_t> buffer_out,
+                       nvme::GenericQueueEntryCmd& nvme_cmd,
+                       Allocation& allocation, uint32_t nsid) {
+  scsi::UnmapCommand unmap_cmd;
```

Comment:
= {} for zero

Reply:
Why is it necessary to zero memory that will be immediately overwritten by readvalue?